src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -332,11 +332,13 @@ public bool InitializeForBatch(TaskLoggingContext loggingContext, ItemBucket bat
                 return false;
             }
 
-            string realTaskAssemblyLoaction = TaskInstance.GetType().Assembly.Location;
-            if (!string.IsNullOrWhiteSpace(realTaskAssemblyLoaction) &&
-                realTaskAssemblyLoaction != _taskFactoryWrapper.TaskFactoryLoadedType.Path)
+            // TODO ask why for net task host it returns false net472\MSBuild\Current\Bin\Microsoft.Build.dll instead of path to a custom task.
+            // Interestingly TaskInstance._taskType contains the correct path.
```

Comment:
@rainersigwald , it's a question for you when you will review :)

Reply:
created a separate ticket to fix it https://github.com/dotnet/msbuild/issues/11954